### PR TITLE
Fix/node v18

### PR DESCRIPTION
**Pull request by y-c-wang**
## Why do we need this PR?
* Node 14 is not supported in the release of [Firebase CLI v12.0.0](https://github.com/firebase/firebase-tools/releases/tag/v12.0.0).
* This PR upgrade the project to support node v18.

## Important Changes
* Migrate from Create-React-App to Vite.
  * Prefix of Environmental Variables Changes from `REACT_APP_` to `VITE_`.
* Firebase update to v8.6.8.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -7,29 +7,33 @@ on:
...
-          REACT_APP_FIREBASE_PROJECT_ID: ${{ secrets.FIREBASE_PROJECT_ID }}
-          REACT_APP_FIREBASE_STORAGE_BUCKET: ${{ secrets.FIREBASE_STORAGE_BUCKET }}
-          REACT_APP_FIREBASE_MESSAGING_SENDER_ID: ${{ secrets.FIREBASE_MESSAGING_SENDER_ID }}
-          REACT_APP_GRAPHQL_API_URL: ${{ secrets.GRAPHQL_API_URL }}
-          REACT_APP_GRAPHQL_WEBSOCKET_URL: ${{ secrets.GRAPHQL_WEBSOCKET_URL }}
-          REACT_APP_SENTRY_DSN: ${{ secrets.REACT_APP_SENTRY_DSN }}
\ No newline at end of file
+      - name: Create Production Build
+        run: yarn build
+      - name: Predeploy to firebase channel
```
Comment (linyc0817) [inline .github/workflows/deploy.yml:33]: Deploy to firebase channel?

**Review by hsinlun99 — COMMENTED** [review]
noted!! 謝謝昱宸!

**Review by hsinlun99 — APPROVED** [review]
approved

**Review by linyc0817 — APPROVED** [review]
Solved

**Pull request merged**